airflow/providers/google/cloud/triggers/bigquery_dts.py

```diff
@@ -0,0 +1,156 @@
+    credentials, or chained list of accounts required to get the access_token
+    of the last account in the list, which will be impersonated in the request.
+    If set as a string, the account must grant the originating account
+    the Service Account Token Creator IAM role.
+    If set as a sequence, the identities from the list must grant
+    Service Account Token Creator IAM role to the directly preceding identity, with first
+    account from the list granting this role to the originating account (templated).
+    """
+
+    def __init__(
+        self,
+        project_id: str,
+        config_id: str,
+        run_id: str,
+        poll_interval: int = 10,
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: str | None = None,
+        location: str | None = None,
+        impersonation_chain: str | Sequence[str] | None = None,
+    ):
+        super().__init__()
+        self.project_id = project_id
+        self.config_id = config_id
+        self.run_id = run_id
+        self.poll_interval = poll_interval
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.location = location
+        self.impersonation_chain = impersonation_chain
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serializes class arguments and classpath."""
+        return (
+            "airflow.providers.google.cloud.triggers.bigquery_dts.BigQueryDataTransferRunTrigger",
+            {
+                "project_id": self.project_id,
+                "config_id": self.config_id,
+                "run_id": self.run_id,
+                "poll_interval": self.poll_interval,
+                "gcp_conn_id": self.gcp_conn_id,
+                "delegate_to": self.delegate_to,
+                "location": self.location,
+                "impersonation_chain": self.impersonation_chain,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        """
+        Get Transfer Run status and if it one of the statuses which mean end of the job
+        then yield TriggerEvent object.
+        """
+        hook = self._get_async_hook()
+        while True:
+            try:
+                transfer_run: TransferRun = await hook.get_transfer_run(
+                    project_id=self.project_id,
+                    config_id=self.config_id,
+                    run_id=self.run_id,
+                )
+                state = transfer_run.state
+                self.log.info("Current state is %s", state)
+
+                if state == TransferState.SUCCEEDED:
+                    self.log.info("Job has completed it's work.")
+                    yield TriggerEvent(
+                        {
+                            "status": "success",
+                            "run_id": self.run_id,
+                            "message": "Job completed",
+                        }
+                    )
+                    return
```

Comment:
return statement here should not require since you are yielding

Reply:
If there is no return statement then trigger logic won't work in python 3.7. Trigger will do it's logic of checking transfer status infinitely and won't stop even after yielding the event. Also source code of the base class says that it is required to return None after yield statement

Reply:
> If there is no return statement then trigger logic won't work in python 3.7. Trigger will do it's logic of checking transfer status infinitely and won't stop even after yielding the event.

this is weird, I was expecting that yield would behave the same with python3.7 and python > python3.7 did you find some docs which have some more insight for why it is behaving differently, I would love to learn it. 

 > Also source code of the base class says that it is required to return None after yield statement

by default Python function return none if you don't return anything won't that would be sufficient?